src/resources/views/buttons/delete.blade.php

```diff
@@ -1,52 +1,55 @@
 
-          if (confirm("{{ trans('backpack::crud.delete_confirm') }}") == true) {
-              $.ajax({
-                  url: delete_url,
-                  type: 'DELETE',
-                  success: function(result) {
-                      // Show an alert with the result
-                      new PNotify({
-                          title: "{{ trans('backpack::crud.delete_confirmation_title') }}",
-                          text: "{{ trans('backpack::crud.delete_confirmation_message') }}",
-                          type: "success"
-                      });
-                      // delete the row from the table
-                      delete_button.parentsUntil('tr').parent().remove();
-                  },
-                  error: function(result) {
-                      // Show an alert with the result
-                      new PNotify({
-                          title: "{{ trans('backpack::crud.delete_confirmation_not_title') }}",
-                          text: "{{ trans('backpack::crud.delete_confirmation_not_message') }}",
-                          type: "warning"
-                      });
-                  }
-              });
-          } else {
-              new PNotify({
-                  title: "{{ trans('backpack::crud.delete_confirmation_not_deleted_title') }}",
-                  text: "{{ trans('backpack::crud.delete_confirmation_not_deleted_message') }}",
-                  type: "info"
-              });
-          }
-        });
+	  function deleteEntry(button) {
+	      // ask for confirmation before deleting an item
+	      // e.preventDefault();
+	      var button = $(button);
+	      var route = button.attr('data-route');
+	      var row = $("#crudTable a[data-route='"+route+"']").parentsUntil('tr').parent();
+
+	      if (confirm("{{ trans('backpack::crud.delete_confirm') }}") == true) {
+	          $.ajax({
+	              url: route,
+	              type: 'DELETE',
+	              success: function(result) {
+	                  // Show an alert with the result
+	                  new PNotify({
+	                      title: "{{ trans('backpack::crud.delete_confirmation_title') }}",
+	                      text: "{{ trans('backpack::crud.delete_confirmation_message') }}",
+	                      type: "success"
+	                  });
+
+	                  // close the modal, if any
+	                  $('.modal').modal('hide');
+
+	                  // delete the row from the table
+	                  row.remove();
+	              },
+	              error: function(result) {
+	                  // Show an alert with the result
+	                  new PNotify({
+	                      title: "{{ trans('backpack::crud.delete_confirmation_not_title') }}",
+	                      text: "{{ trans('backpack::crud.delete_confirmation_not_message') }}",
+	                      type: "warning"
+	                  });
+	              }
+	          });
+	      } else {
+	          new PNotify({
```

Comment:
What does this do? 🕋

Reply:
See? You wouldn't know without the comment! :-)) Added comment here too.